lib/utils/source_ip_check.ts

```diff
@@ -116,25 +118,36 @@ export function checkSourceIpForHandler<P1, P2, P3, P4, P5, P6, O>(
   p4: P4,
   p5: P5,
   p6: P6
-) => Promise<O | IResponseErrorForbiddenNotAuthorized> {
+) => Promise<
+  O | IResponseErrorForbiddenNotAuthorized | IResponseErrorInternal
+> {
   return (p1: P1, p2: P2, p3: P3, p4: P4, p5: P5, p6: P6) => {
-    // extract the x-forwarded-for header and the allowed cidrs from the params
-    const x = extractor(p1, p2, p3, p4, p5, p6);
-    const maybeClientIp = x.e1;
-    const cidrs = x.e2;
+    return new Promise(resolve => {
+      // extract the x-forwarded-for header and the allowed cidrs from the params
+      const x = extractor(p1, p2, p3, p4, p5, p6);
+      const maybeClientIp = x.e1;
+      const cidrs = x.e2;
 
-    if (
-      // we have a client IP and...
-      isSome(maybeClientIp) &&
-      // ...either no CIDRs setting or client IP is contained in allowed CIDRs
-      (cidrs.size === 0 || isContainedInCidrs(maybeClientIp.value, cidrs))
-    ) {
-      // forward request to handler
-      return handler(p1, p2, p3, p4, p5, p6);
-    } else {
-      // respond with Not Authorized
-      return Promise.resolve(ResponseErrorForbiddenNotAuthorized);
-    }
+      if (isNone(maybeClientIp)) {
+        return resolve(
+          ResponseErrorInternal(
+            "IP address cannot be extracted from the request, check x-forwarder-for HTTP header"
```

Comment:
this error will reach the client, but the client doesn't know anything about `x-forwarded-for`...

Reply:
also this step should not run if `cidrs.size === 0`

Reply:
ok

Reply:
just notice that the actual behavior is to fail when the IP is not found (even if cidr.size == 0)

https://github.com/teamdigitale/digital-citizenship-functions/blob/master/lib/utils/source_ip_check.ts#L128

Reply:
obviously I'm not convinced about either option :D what's your opinion?

Reply:
makes sense but I would link that behavior only to a non-production environment (i.e.  `NODE_ENV !== "production"`)